src/Util/ClassSourceManipulator.php

```diff
@@ -305,14 +310,16 @@ public function createMethodLevelBlankLine()
         return $this->createBlankLineNode(self::CONTEXT_CLASS_METHOD);
     }
 
-    public function addProperty(string $name, array $annotationLines = [], $defaultValue = null)
+    public function addProperty(string $name, array $annotationLines = [], $defaultValue = null, $type = null)
     {
         if ($this->propertyExists($name)) {
             // we never overwrite properties
             return;
         }
 
         $newPropertyBuilder = (new Builder\Property($name))->makePrivate();
+        null == $type ?: $newPropertyBuilder->setType(new Node\NullableType($type));
```

Comment:
I think it'd be less confusing to have an actual condition there, instead of that inline ternary shorthand. 

```suggestion
        if (null !== $type) {
            $newPropertyBuilder->setType(new Node\NullableType($type));
        }
```

Either way, use the strict equality operator instead of `==`:
```suggestion
        null === $type ?: $newPropertyBuilder->setType(new Node\NullableType($type));
```